src/js/autocomplete.js

```diff
@@ -6,9 +13,187 @@ class Autocomplete {
 	 */
 	constructor (autocompleteEl, options) {
 		this.autocompleteEl = autocompleteEl;
-		this.options = Object.assign({}, {
+		this.options = Object.assign({
+			placeholder: '',
+			cssNamespace: 'o-autocomplete',
+			displayMenu: 'overlay',
+			showNoOptionsFound: false,
+			templates: {
+				suggestion: this.suggestionTemplate.bind(this)
+			}
 		}, options || Autocomplete.getDataAttributes(autocompleteEl));
+
+		// If highlighter is a string, then it is the name of a global function to use.
+		// If highlighter is not a string, then it is a function to use.
+		if (typeof this.options.highlighter === 'string') {
+			this.options.highlighter = window[this.options.highlighter];
+		}
+
+		const container = document.createElement('div');
+		this.container = container;
+		container.classList.add('o-autocomplete__container');
+
+		if (this.options.source) {
+			// If source is a string, then it is the name of a global function to use.
+			// If source is not a string, then it is a function to use.
+			/**
+			 * @function
+			 * @param {string} query - Text which was typed into the autocomplete by the user
+			 * @param {PopulateResults} populateResults - Function to call when ready to update the suggestions dropdown
+			 * @returns {void}
+			 */
+			const customSuggestions = typeof this.options.source === 'string' ? window[this.options.source] : this.options.source;
+			this.options.source = function(query, populateResults) {
+				showLoadingPane();
+				const callback = function(results) {
+					hideLoadingPane();
+					populateResults(results);
+				};
+				customSuggestions(query, callback);
+			};
+			autocompleteEl.innerHTML = '';
+			autocompleteEl.appendChild(container);
+			const id = autocompleteEl.getAttribute('id');
+			autocompleteEl.removeAttribute('id');
```

Comment:
The reason we use the same `id` is because it could be linked to from a `label` element's `for` attribute. If we generate a new `id`, I think we would then need to find if a label exists which has a `for` attribute with the old `id` and update the attribute to use the new `id`?

Reply:
Oh I see, good point! In that case the the `id` should be set on the fallback input instead, so it's connected to the label if js fails.

```diff
-            <div data-o-component="o-autocomplete" data-o-autocomplete-source="customSuggestions" data-o-autocomplete-highlighter="customHighlighter" class="o-autocomplete"  id="my-autocomplete">
+            <div data-o-component="o-autocomplete" data-o-autocomplete-source="customSuggestions" data-o-autocomplete-highlighter="customHighlighter" class="o-autocomplete" >
                <!-- If the JavaScript executes, then this input will be progressively enhanced to an autocomplete component -->
-                <input required type="text" name="text-example">
+                <input required type="text" name="text-example" id="my-autocomplete">
            </div>
```

Reply:
Would be a good thing to shout about in the readme too :)

Reply:
Done in [`eb54004` (#2)](https://github.com/Financial-Times/o-autocomplete/pull/2/commits/eb54004c313d90eeb5b012e8bb8eb61f422d6857)